content/en/users/tutorials/oidc-agent-fedcloudclient-terraform/_index.md

```diff
@@ -0,0 +1,510 @@
+             eduperson_scoped_affiliation \
+             eduperson_unique_id" egi
+# Listing existing configuration
+$ oidc-add -l
+# Requesting an OIDC access token
+$ oidc-token egi
+# Exporting a variable with a Check-in OIDC access token to be used with OpenStack
+# XXX access tokens are short lived, relaunch command to obtain a new token
+$ export OS_ACCESS_TOKEN=`oidc-token egi`
+```
+
+It's possible to automatically start `oidc-agent` in your shell initialisation,
+example that can be added to `~/.bash_profile` or `~/.zshrc`:
+
+```bash
+if command -v oidc-agent-service &> /dev/null
+  eval `oidc-agent-service use`
+  # for fedcloudclient, once egi account got created
+  export OIDC_AGENT_ACCOUNT=egi
+fi
+```
+
+When using `oidc-agent-service`,
+[fedcloudclient](#installing-fedcloudclient-and-ansible) will be able to
+automatically request a new access token from `oidc-agent`.
+
+See [full documentation](https://indigo-dc.gitbook.io/oidc-agent/).
+
+### Installing fedcloudclient and ansible
+
+[`fedcloudclient`](https://fedcloudclient.fedcloud.eu/) is an high-level Python
+package for a command-line client designed for interaction with the OpenStack
+services in the EGI infrastructure. The client can access various EGI services
+and can perform many tasks for users including managing access tokens, listing
+services, and mainly execute commands on OpenStack sites in EGI infrastructure.
+
+`fedcloudclient` can leverage [`oidc-agent`](#setting-up-oidc-agent) if it's
+installed and properly configured.
+
+`fedcloudclient` and
+[`opentackclient`](https://docs.openstack.org/python-openstackclient/latest/),
+the official OpenStack python client, will be used to interact with the EGI
+Cloud Compute service.
+
+Required python dependencies are documented in a `requirements.txt` file
+(Ansible will be used at a later stage, but is installed at the same time):
+
+```requirements.txt
+openstackclient
+fedcloudclient
+ansible
+```
+
+For keeping the main system tidy and isolating the environment, the python
+packages will be installed in a dedicated
+[python virtualenv](https://docs.python.org/3/tutorial/venv.html):
+
+```shell
+# Creating an arbitrary directory where to store python virtual environments
+$ mkdir -p ~/.virtualenvs
+# Creating a python 3 virtual environment
+$ python3 -m venv ~/.virtualenvs/fedcloud
+# Activating the virtual environment
+$ source ~/.virtualenvs/fedcloud
+# Installing required python packages in the virtual environment
+$ pip install -r requirements.txt
+```
+
+### Identifying a suitable cloud site
+
+It's possible to deploy an OpenStack Virtual Machine (VM) on any of the sites
+supporting the Virtual Organisations (VO) you are a member of.
+
+Once [fedcloudclient](#installing-fedcloudclient-and-ansible) is installed it's
+possible to get information about the OIDC token accessed via
+[oidc-agent](#setting-up-oidc-agent).
+
+```shell
+# Verifying that an OS_ACCESS_TOKEN is available
+$ echo $OS_ACCESS_TOKEN
```

Comment:
why not just using the oidc agent support in fedcloudclient?

Reply:
You are right, I removed this and marked exporting `OS_ACCESS_TOKEN` as an example.